src/plugins/category/plugin.js

```diff
@@ -0,0 +1,470 @@
+                    }
+                    else {
+                        model = self.addGroup(group, false, item.data, self.parseGroupFlags(item));
+                        if (model === null) {
+                            return;
+                        }
+
+                        self.applyGroupFlags(model);
+
+                        add(item, model);
+                    }
+                }
+                else {
+                    if (!item.empty) {
+                        if (item.id === undefined) {
+                            Utils.error(!options.allow_invalid, 'RulesParse', 'Missing rule field id');
+                            item.empty = true;
+                        }
+                        if (item.operator === undefined) {
+                            item.operator = 'equal';
+                        }
+                    }
+
+                    model = self.addRule(group, item.data, self.parseRuleFlags(item));
+                    if (model === null) {
+                        return;
+                    }
+
+                    if (!item.empty) {
+                        if (item.category){
+                            model.category = self.getCategoryById(item.category ? item.category : '-1');
+                            if (self.isFilterOKForCategory(item.id, item.category)){
+
+                                model.filter = self.getFilterById(item.id, !options.allow_invalid);
+
+                                if (model.filter) {
+                                    model.operator = self.getOperatorByType(item.operator, !options.allow_invalid);
+
+                                    if (!model.operator) {
+                                        model.operator = self.getOperators(model.filter)[0];
+                                    }
+
+                                    if (model.operator && model.operator.nb_inputs !== 0 && item.value !== undefined) {
+                                        model.value = item.value;
+                                    }
+                                }
+                            }
+                        }
+                    }
+
+                    self.applyRuleFlags(model);
+
+                    if (self.change('jsonToRule', model, item) != model) {
+                        Utils.error('RulesParse', 'Plugin tried to change rule reference');
+                    }
+                }
+            });
+
+            if (self.change('jsonToGroup', group, data) != group) {
+                Utils.error('RulesParse', 'Plugin tried to change group reference');
+            }
+
+        }(data, this.model.root));
+    },
+
+    /**
+     * Decide if filter is available for category
+     * @param filterId {string}
+     * @param categoryId {string}
+     * @return {boolean} true if filter belongs to that category
+     */
+    isFilterOKForCategory: function(filterId, categoryId) {
+        var filtersOfCategory = this.getFiltersForCategoryId(categoryId);
```

Comment:
useless access to `getFiltersForCategoryId`: use `getCategoryById` and loop over its `filters`